packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -239,4 +211,23 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
     });
     forceUpdate();
   }, [apiRef, setGridState, forceUpdate, isRowSelectable]);
+
+  React.useEffect(() => {
+    apiRef.current.registerControlState<GridSelectionModel, GridSelectionState>({
```

Comment:
"register" would suggest it's run one and should be cleanup, It's not what happens.

```suggestion
    apiRef.current.updateControlState<GridSelectionModel, GridSelectionState>({
```

Reply:
Oh boy, it's does indeed register. This can't work.

Reply:
It runs once until you change the props. 
I think register is great as you tell the state that this prop and onChange are controlled. You don't update the state itself... Up to you!

Reply:
My issue with "register" is that it implicitly suggests we should also unregister to cleanup, which we don't. It seems to be a setter.